### PR TITLE
connectivity: Introdue Multicast connectivity test

### PR DESCRIPTION
**Pull request by yushoyamaguchi**
Introduces Multicast connectivity test, working with netshoot container.
Using socat for IGMP and UDP communication test.
Receivers are running as Daemonset and a sender is running as a Deployment which has 1 replicaset.

Operation was confirmed on the kind and kubeadm clusters.

This if follow up of https://github.com/cilium/cilium-cli/issues/2615 .

In addtion, this PR is related to https://github.com/cilium/cilium-cli/pull/2620 .
I want to implement a test scenario using multicast-subcommand after this is merged.

In addition, I'd like to put this test into GitHub CI.

<details>
<summary>related part of this command output (when 3 nodes)</summary>

```
[=] [cilium-test-1] Test [multicast] [100/103]
  [.] Action [multicast/multicast/socat multicast]
  🐛 Executing command [timeout 1 socat STDIO UDP4-RECVFROM:6666,ip-add-membership=239.255.9.9:10.244.0.10]
.  [.] Action [multicast/multicast/socat multicast]
  🐛 Executing command [timeout 1 socat STDIO UDP4-RECVFROM:6666,ip-add-membership=239.255.9.9:10.244.2.142]
.  [.] Action [multicast/multicast/socat multicast]
  🐛 Executing command [timeout 1 socat STDIO UDP4-RECVFROM:6666,ip-add-membership=239.255.9.9:10.244.1.88]
.  🐛 Finalizing Test multicast
  🐛 Finalizing Test host-firewall-ingress
  🐛 Finalizing Test host-firewall-egress
  🐛 Finalizing Test check-log-errors

✅ [cilium-test-1] All 1 tests (3 actions) successful, 102 tests skipped, 0 scenarios skipped.
```

</details>

### CODE REVIEW

File: cilium-cli/connectivity/check/context.go
Comment (yushoyamaguchi) [inline cilium-cli/connectivity/check/context.go]: I think it may be good to make waiting-time longer. (Currently 1 sec)
The reasons are
1. This command exits with exit-code 0 WITHOUT waiting for a second after receiving a packet.
2. If multicast configuration and implementation are correct, all pods in which this command is executed will receive multicast packet soon

This waiting-time will only be applied when there is an issue with the multicast implementation or configuration.

I don't think there's a way to determine within a limited time if packets are not being received other than using a timeout, so I believe it's difficult to eliminate the timeout altogether.

File: cilium-cli/connectivity/check/context.go
Comment (yushoyamaguchi) [inline cilium-cli/connectivity/check/context.go:1243]: `for i in $(seq 1 10000); do echo "%s" | socat - UDP-DATAGRAM:%s:%s; sleep 0.1; done`
Parameters of this command might need to be changed.

I want to ensure that packets are reliably received during the above-mentioned wait-time (currently 1 sec). 
With the current parameters, it feels like too many packets are being sent unnecessarily...

**Comment by yushoyamaguchi** [comment]
@dylandreimerink 
Thank you for permitting the integrated test execution.

Is the cause of this error in my code? (I don't understand what happened. I'm sorry...)
https://github.com/cilium/cilium/actions/runs/10828109927/job/30042735033

Maybe, this is the cause...
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/identity/cache   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
```

**Comment by yushoyamaguchi** [comment]
@christarazi @asauber @dylandreimerink @tommyp1ckles @ldelossa 

I'm sorry for mentioning.
Is it difficult to review the PR for merging?

**Comment by dylandreimerink** [comment]
I can't review it since I am not part of the codeowners for this PR, I am merely doing the admin like setting labels and triggering tests for which you need some privileges. I hope @christarazi and @asauber can find some time to review it, otherwise we can request alternative reviewers from the respective teams.

File: cilium-cli/connectivity/check/context.go
Comment (asauber) [reply cilium-cli/connectivity/check/context.go]: A timeout does seem like the right failure condition for this test.

Do you have any concerns that there will be false negatives (failures), with a timeout of only 1 second?

I think 5 seconds should be fine here.

File: cilium-cli/connectivity/check/context.go
Comment (asauber) [reply cilium-cli/connectivity/check/context.go:1243]: Sending a packet every 100ms from each pod seems fine to me.

**Review by asauber — CHANGES_REQUESTED** [review]
The implementation here looks fine to me. It's mostly in line with our other connectivity tests, in particular the BGP tests.

Would you be able to do some interactive testing and confirm that we can use the `alpine/socat` container image rather than the `nicolaka/netshoot` container for these tests?

```
$ docker save docker.io/nicolaka/netshoot:v0.13@sha256:a20c2531bf35436ed3766cd6cfe89d352b050ccc4d7005ce6400adf97503da1b -o netshoot.tar

$ docker save docker.io/alpine/socat:1.7.4.4@sha256:93efcf633c5489b170404df25e289f811cabeea728a5367f0c9b1560982edf14 -o alpinesocat.tar

$ ls -la
total 551860
drwxrwxr-x  2 ubuntu ubuntu      4096 Oct  7 16:50 .
drwxr-x--x 17 ubuntu ubuntu      4096 Oct  7 16:49 ..
-rw-------  1 ubuntu ubuntu   8908288 Oct  7 16:50 alpinesocat.tar
-rw-------  1 ubuntu ubuntu 556183552 Oct  7 16:49 netshoot.tar
```

The container image size for netshoot is 62x larger, which can significantly affect the speed of the test.

File: cilium-cli/connectivity/check/context.go
Comment (yushoyamaguchi) [reply cilium-cli/connectivity/check/context.go]: I've changed.

**Review by asauber — APPROVED** [review]

**Comment by yushoyamaguchi** [comment]
@christarazi 
I'm sorry for mentioning.
Could you review?

**Review by christarazi — APPROVED** [review]
Apologies for the delay.

This LGTM assuming that the multicast code only has a command-line level access to manipulate state. Ideally, moving forward the Cilium CLI connectivity tests are meant not to manipulate state of Cilium or run subsequent commands to retrieve state in the middle of the test, as it could be an additional source of flakes. For now, this is ok though.

**Comment by yushoyamaguchi** [comment]
@christarazi 
Thank you for approval.

I understand that while I have separated the namespaces, it would be better to increase the level of isolation. However, at this point, I couldn't come up with a way to create a more isolated environment. I apologize

Could you allow the remaining tests?

**Pull request merged**
